stdlib/src/utils/string_slice.mojo

```diff
@@ -427,17 +427,15 @@ struct StringSlice[is_mutable: Bool, //, origin: Origin[is_mutable]](
         Returns:
             If the `StringSlice` is equal to the input in length and contents.
         """
-        if not self and not rhs:
-            return True
-        if len(self) != len(rhs):
+
+        var s_len = self.byte_length()
+        s_ptr, rhs_ptr = self.unsafe_ptr(), rhs.unsafe_ptr()
```

Comment:
**Suggestion** Please split these assignments into separate lines and use `var` for each.